main.py

```diff
@@ -1,17 +1,271 @@
 """
 This is a starter file to get you going. You may also include other files if you feel it's necessary.
-
 Make sure to follow the code convention described here:
 https://github.com/UWARG/computer-vision-python/blob/main/README.md#naming-and-typing-conventions
-
 Hints:
 * The internet is your friend! Don't be afraid to search for tutorials/intros/etc.
 * We suggest using a convolutional neural network.
 * TensorFlow Keras has the CIFAR-10 dataset as a module, so you don't need to manually download and unpack it.
 """
 
-# Import whatever libraries/modules you need
+# Importing Libraries
+import torch
+import torchvision
+import torchvision.transforms as transforms
+from torch.utils.data import random_split
+import torch.nn as nn
+import matplotlib.pyplot as plt
+
+# Normalising testing and training data [-1,1], this ensures same scale is used for all data (faster convergence)
+# Applying randomness noise with cropping and horizontal flipping to training transform
+data_statistics = ((0.5, 0.5, 0.5), (0.5, 0.5, 0.5))
+train_transforms_cifar = transforms.Compose([
+    transforms.RandomCrop(32, padding=4, padding_mode='reflect'),
+    transforms.RandomHorizontalFlip(),
+    transforms.ToTensor(),  # CxHxW
+    transforms.Normalize(*data_statistics, inplace=True)  # [-1,1], data = (data-mean)/std_dev
+])
+test_transforms_cifar = transforms.Compose([
+    transforms.ToTensor(),  # CxHxW
+    transforms.Normalize(*data_statistics, inplace=True)  # [-1,1], data = (data-mean)/std_dev
+])
+
+# Storing the train and testing datasets, determined from 'bool train', applying transformations
+dataset = torchvision.datasets.CIFAR10(root="data/", download=True, transform=train_transforms_cifar)
+test_dataset = torchvision.datasets.CIFAR10(root="data/",
+                                            download=True, train=False, transform=test_transforms_cifar)
+# Creating datasets for training and validation
+# The validation data will help eliminate data-memorizing, and will make up 20% of the training data
+val_ratio = 0.2
+train_dataset, val_dataset = random_split(dataset, [int((1-val_ratio)*len(dataset)), int(val_ratio * len(dataset))])
+
+# Creating dataloading wrappers for the 3 datasets, used to access data more efficiently
+batch_size = 32
+train_dl = torch.utils.data.DataLoader(train_dataset, batch_size, shuffle=True, pin_memory=True)
+val_dl = torch.utils.data.DataLoader(val_dataset, batch_size, shuffle=True, pin_memory=True)
+test_dl = torch.utils.data.DataLoader(test_dataset, batch_size, pin_memory=True)
+
+
+# Determines whether the machine is Cuda enabled, allowing GPU access
+def get_default_device():
```

Comment:
Nit: It is good practice to also specify what the return type will be for the function. It helps make the code more readable.